src/lib/components/Tabs/index.js

```diff
@@ -109,78 +200,67 @@ class Tabs extends BaseElement {
     switch (e.keyCode) {
       case KEYCODE.RIGHT:
         e.preventDefault();
-        this.focusNextItem();
+        this.nextTab();
         break;
       case KEYCODE.LEFT:
         e.preventDefault();
-        this.focusPreviousItem();
+        this.previousTab();
         break;
       case KEYCODE.HOME:
         e.preventDefault();
-        this.focusFirstItem();
+        this.firstTab();
         break;
       case KEYCODE.END:
         e.preventDefault();
-        this.focusLastItem();
+        this.lastTab();
         break;
     }
+    tabs[this.activeTab].focus();
   }
 
-  // Figure out if the current element has a next sibling.
-  // If so, move focus to it.
-  focusNextItem() {
-    const item = document.activeElement;
-    if (item.nextElementSibling) {
-      this.changeTab(item.nextElementSibling);
+  // Helper method to allow other components to focus an arbitrary tab.
+  focusTab(index) {
+    const tabs = this.querySelectorAll(".web-tabs__tab");
+
+    if (!tabs[index]) {
+      throw new RangeError("There is no tab at the specified index.");
     } else {
-      this.focusFirstItem();
+      this.activeTab = index;
     }
   }
 
-  // Figure out if the current element has a previous sibling.
-  // If so, moving focus to it.
-  focusPreviousItem() {
-    const item = document.activeElement;
-    if (item.previousElementSibling) {
-      this.changeTab(item.previousElementSibling);
+  // If previous tab exists, make it active. If not, make last tab active.
+  previousTab() {
+    const tabs = this.querySelectorAll(".web-tabs__tab");
+
+    if (tabs[this.activeTab - 1]) {
+      this.activeTab = this.activeTab - 1;
     } else {
-      this.focusLastItem();
+      this.activeTab = tabs.length - 1;
     }
   }
 
-  // Focus first element in set of siblings.
-  focusFirstItem() {
-    const item = document.activeElement;
-    this.changeTab(item.parentElement.firstElementChild);
-  }
+  // If next tab exists, make it active. If not, make first tab active.
+  nextTab() {
+    const tabs = this.querySelectorAll(".web-tabs__tab");
 
-  // Focus last element in set of siblings.
-  focusLastItem() {
-    const item = document.activeElement;
-    this.changeTab(item.parentElement.lastElementChild);
+    if (tabs[this.activeTab + 1]) {
```

Comment:
```js
this.activeTab = (this.activeTab + 1) % tabs.length;

// or to prevent NaN if there's no tabs
this.activeTab = ((this.activeTab + 1) % tabs.length) || 0;
```